spec/services/is_eligible_for_international_relocation_payment_spec.rb

```diff
@@ -15,54 +15,83 @@
     let(:course_subject) { create(:subject, name: 'Quantum Mechanics', code: 'F0') }
 
     context 'application meets all criteria' do
-      it { is_expected.to be true }
+      it 'returns expected value' do
+        if FeatureFlag.active? :hide_international_relocation_payment
+          expect(eligibility).to be false
+        else
+          expect(eligibility).to be true
+        end
+      end
     end
 
     context 'right_to_work_or_study set to decide later' do
       let(:right_to_work_or_study) { 'decide_later' }
 
-      it { is_expected.to be true }
+      it 'returns expected value' do
+        if FeatureFlag.active? :hide_international_relocation_payment
+          expect(eligibility).to be false
+        else
+          expect(eligibility).to be true
+        end
+      end
     end
 
     context 'subject is physics' do
       let(:course_subject) { create(:subject, name: 'Advanced Quantum Computing', code: 'F0') }
 
-      it { is_expected.to be true }
+      it 'returns expected value' do
+        if FeatureFlag.active? :hide_international_relocation_payment
+          expect(eligibility).to be false
+        else
+          expect(eligibility).to be true
+        end
+      end
     end
 
     context 'subject is modern foreign languages' do
       let(:course_subject) { create(:subject, name: 'Russian', code: '21') }
 
+      # Always false, regardless of feature flag
       it { is_expected.to be false }
     end
 
     context 'subject is french' do
       let(:course_subject) { create(:subject, name: 'French', code: '15') }
 
-      it { is_expected.to be true }
+      it 'returns expected value' do
+        if FeatureFlag.active? :hide_international_relocation_payment
+          expect(eligibility).to be false
+        else
+          expect(eligibility).to be true
+        end
+      end
```

Comment:
Can we avoid if statement on this specs? I'd recommend to use the feature flag meta description on specs to activate (`feature_flag: :some_feature_flag` to activate a feature flag for a test). And we want the feature flag to be off, then calls deactivate directly.

Reply:
We run all the tests twice, with feature flags on and then with feature flags off, right? So if we force them on / off each time, running them twice is pointless. This ensures both scenarios are tested.